### PR TITLE
Milestone 5b - Friendship

### PR DESCRIPTION
**Pull request by Laguna1**
A 'Friendship' model with associations and all requested features for friendships have been created.
Added tests.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -6,4 +6,30 @@
...
+
+  <ol>
+  <% @user.friends.each do |friend| %>
+    <li>
+      <%= friend.name %>
+      <% if @user == current_user %>
+        <%= link_to 'Unfriend',
+        friendships_destroy_path(friendship_id: Friendship.find_friendship(@user.id, friend.id))  %>
+      <% end %>
+    </li>
```
Comment (maelfosso) [inline app/views/users/show.html.erb]: Remove logic from your views. Please, move them in view helpers.

File: app/controllers/friendships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  def destroy
+    friendship = Friendship.find(params[:friendship_id])
+    friendship.destroy
+    redirect_to user_path(current_user.id)
+  end
+
+  def update
+    friendship = Friendship.find(params[:friendship_id])
+    p current_user
```
Comment (maelfosso) [inline app/controllers/friendships_controller.rb:18]: Kindly remove it, please.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for applying the suggested changes
However, there are some missing requirements

1. There should be a button to invite user to friendship on the single user page, as indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png), Kindly add it

2. After sending a friendship, there must be a button to cancel it.

3. There is a problem with the friendship feature. 
    - Without accepting the friendship, The two users can see the posts of each other
    - Also, On the second user side, there  is no button for accepting the friendship request with the friendship row with this
```rb
2.7.0 :006 > Friendship.all
  Friendship Load (0.8ms)  SELECT  "friendships".* FROM "friendships" LIMIT $1  [["LIMIT", 11]]
 => #<ActiveRecord::Relation [#<Friendship id: 1, user_id: 2, friend_id: 1, confirmed: false, created_at: "2020-10-01 22:01:32", updated_at: "2020-10-01 22:01:32">]> 
2.7.0 :007 >
```
All those points are not OK. Kindly fix that.

4. There is no integration tests. Kindly add them as required [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png)

5. Remove logic from your views. Please, move them in view helpers.

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

File: app/views/users/show.html.erb (hunk 1 above)
Comment (chinweokwu) [reply app/views/users/show.html.erb]: completed the 1, 4, 5 reviews made by tse. 2 & 3 are not part of the microverse requirments.

Comment (chinweokwu) [reply app/views/users/show.html.erb]: completed the 1, 4, 5 reviews made by tse. 2 & 3 are not part of the microverse requirements.

File: app/controllers/friendships_controller.rb (hunk 2 above)
Comment (chinweokwu) [reply app/controllers/friendships_controller.rb:18]: it cannot be removed. it update friends when the user clicks on accepts so i can't remove it

**Review by lucasmazo32 — APPROVED** [review]
# Project approved :rocket:

#### Great job!

#### Your project looks great!

### You can go ahead and merge this pull request!

### Happy coding! :computer:

## Lucas Mazo

**Pull request merged**